src/shell/components/FieldTypeUUID/FieldTypeUUID.js

```diff
@@ -18,39 +23,87 @@ export const FieldTypeUUID = React.memo(function FieldTypeUUID(props) {
     }
   }, []);
 
+  // Set isCopied to false after 5 seconds
+  useEffect(() => {
+    if (isCopied) {
+      const timeout = setTimeout(() => {
+        setIsCopied(false);
+      }, 5000);
+      // Clear timeout if the component is unmounted
+      return () => clearTimeout(timeout);
+    }
+  }, [isCopied]);
+
   return (
     <label className={cx(styles.FieldTypeUUID, props.className)}>
-      <div className={styles.DateFieldTypeInput}>
-        <FontAwesomeIcon
-          className={styles.Icon}
-          icon={faClipboard}
-          aria-hidden="true"
-          title="Click to Copy"
-          onClick={(e) => {
-            const input = document.createElement("input");
-            document.body.appendChild(input);
-            input.value = props.value;
-            input.focus();
-            input.select();
-            const result = document.execCommand("copy");
-            input.remove();
-            if (result === "unsuccessful") {
-              return props.dispatch(
-                notify({
-                  type: "error",
-                  message: "Failed to copy the team ID to your clipboard",
-                })
-              );
-            }
+      <div>
+        <Tooltip
+          slotProps={{
+            popper: {
+              modifiers: [
+                {
+                  name: "offset",
+                  options: {
+                    offset: [0, -47],
+                  },
+                },
+              ],
+            },
           }}
-        />
+          placement="top"
+          title="This field cannot be edited"
+        >
+          <TextField
+            required={props.required}
+            value={props.value || ""}
+            readOnly={true}
```

Comment:
TextField doesn't have a readOnly prop

Reply:
@theofficialnar  that make sense, didn't noticed it usually it would throw an error but the file is not TS so yeah. Thanks for pointing that.